core/src/main/scala/org/apache/spark/metrics/sink/StatsdSink.scala

```diff
@@ -46,23 +65,51 @@ private[spark] class StatsdSink(
     securityMgr: SecurityManager)
   extends Sink with Logging {
   import StatsdSink._
+  import collection.JavaConverters._
+
+  val reporterFactory = new StatsdReporterFactory()
 
-  val host = property.getProperty(STATSD_KEY_HOST, STATSD_DEFAULT_HOST)
-  val port = property.getProperty(STATSD_KEY_PORT, STATSD_DEFAULT_PORT).toInt
+  reporterFactory.host = property.getProperty(STATSD_KEY_HOST, STATSD_DEFAULT_HOST)
+  reporterFactory.port = property.getProperty(STATSD_KEY_PORT, STATSD_DEFAULT_PORT).toInt
+  reporterFactory.prefix = property.getProperty(STATSD_KEY_PREFIX, STATSD_DEFAULT_PREFIX)
 
   val pollPeriod = property.getProperty(STATSD_KEY_PERIOD, STATSD_DEFAULT_PERIOD).toInt
   val pollUnit =
     TimeUnit.valueOf(property.getProperty(STATSD_KEY_UNIT, STATSD_DEFAULT_UNIT).toUpperCase)
 
-  val prefix = property.getProperty(STATSD_KEY_PREFIX, STATSD_DEFAULT_PREFIX)
+  val excludes = property.getProperty(STATSD_KEY_EXCLUDES, STATSD_DEFAULT_EXCLUDES)
+  if(!excludes.isEmpty()) {
+    reporterFactory.setExcludes(ImmutableSet.copyOf(excludes.split(",")))
+  }
+
+  val includes = property.getProperty(STATSD_KEY_INCLUDES, STATSD_DEFAULT_INCLUDES)
+  if(!includes.isEmpty()) {
+    reporterFactory.setIncludes(ImmutableSet.copyOf(includes.split(",")))
+  }
+
+  val excludesAttributes = property.getProperty(STATSD_KEY_EXCLUDES_ATTRIBUTES, STATSD_DEFAULT_EXCLUDES_ATTRIBUTES).toUpperCase
+  if(!excludesAttributes.isEmpty()) {
+    reporterFactory.setExcludesAttributes(EnumSet.copyOf(excludesAttributes.split(",").map(attr => MetricAttribute.valueOf(attr)).toList.asJava))
```

Comment:
let's extract `Set` parsing/handling functionality to a separate method to keep the code DRY

Reply:
Moved it to separate method.